nodeipam: add a Service LoadBalancer node ipam (#30038)
cilium/cilium: Documentation/network/node-ipam.rst

.. only:: not (epub or latex or html)

    WARNING: You are looking at unreleased Cilium documentation.
    Please use the official rendered version released here:
    https://docs.cilium.io

.. _node_ipam:

************
Node IPAM LB
************

> Node IPAM LoadBalancer is a feature inspired by k3s "ServiceLB" and allow you
> to "advertise" directly the node's IPs inside a Service LoadBalancer. This feature

Reviewer: ```suggestion
Node IPAM LoadBalancer is a feature inspired by k3s "ServiceLB" that allows you
to "advertise" the node's IPs directly inside a Service LoadBalancer. This feature
```

is especially useful if you don't control the network you are running on and can't
use either the L2 or BGP capabilities of Cilium.

> It works by getting the Node addresses of the selected pods and advertise them.

Reviewer: ```suggestion
It works by getting the Node addresses of the selected pods and advertising them.
```

It will respect the ``.spec.ipFamilies`` to decide if IPv4 or IPv6 addresses
shall be used and will use the ``ExternalIP`` addresses if any or the
``InternalIP`` addresses otherwise.

> To use this feature your service must be of type ``LoadBalancer`` and has the

Reviewer: ```suggestion
To use this feature your service must be of type ``LoadBalancer`` and have the
```
